dali/ft/filecopy.cpp

```diff
@@ -3771,7 +3774,21 @@ void FileSprayer::updateTargetProperties()
     if (distributedSource)
     {
         if (distributedSource->querySuperFile()==nullptr)
-            distributedSource->addAttrValue("@numDiskReads", totalNumReads);
+        {
+            IPropertyTree & fileAttr = distributedSource->queryAttributes();
+            cost_type readCost = 0;
+            // Check if it is a legacy file without @readCost attribute -> calculate prevReadCost
+            if (!hasReadWriteCostFields(&fileAttr) && fileAttr.hasProp(getDFUQResultFieldName(DFUQRFnumDiskReads)))
+            {
+                stat_type totalReads = fileAttr.getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads), 0) + totalNumReads;
+                readCost = money2cost_type(calcFileAccessCost(distributedSource, 0, totalReads));
+            }
+            else
+                readCost = money2cost_type(calcFileAccessCost(distributedSource, 0, totalNumReads));
```

Comment:
question: Is this tied down to a single cluster by this point?  Worth defining an expensive plane and a cheap plane in our standard config, and then checking the costs is recorded correctly depending on the source.

Reply:
The decision to assume that the source was from the first cluster was made as part of HPCC-27388.  I recall that there was some discussion and it was agreed that it would be good enough to assume it's from the first cluster.  And I think the basis for that decision was that it was assume that the cost of each plane would be the same.  We can see that this assumption may need to revisited. However, this PR was supposed to about storing the read/write cost to ensure that the keyed join read cost calculations are not overwritten by the legacy dynamic calculations.  I think it would be worth considering this issue in separate jira:https://track.hpccsystems.com/browse/HPCC-30946

Reply:
Agreed.  I think that at some point the clusters are reordered so that the "best" cluster comes first - but I'm not sure if that will be done at this point.  Something to investigate further.